utils/sample_qc.py

```diff
@@ -364,82 +364,139 @@ def infer_sex(
+    logger.info(f'YY {yy_y_ploidy_cutoff}')
 
     sex_expr = hl.struct(
-        sex_karyotype=(
+        X_karyotype=(
             hl.case()
                 .when(
-                (chr_x_ploidy > xx_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xx_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (f_stat < f_stat_female_cutoff),
-                'XX')
+                    (ht.chrX_ploidy < xy_x_ploidy_cutoff),
+                    'X')
+                .when(
+                    ((ht.chrX_ploidy > xx_x_ploidy_cutoffs[0]) &
+                    (ht.chrX_ploidy < xx_x_ploidy_cutoffs[1])),
+                    'XX')
                 .when(
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > xy_y_ploidy_cutoffs[0]) &
-                (chr_y_ploidy < xy_y_ploidy_cutoffs[1]) &
-                (f_stat > f_stat_male_cutoff),
-                'XY')
+                    (ht.chrX_ploidy >= xxx_x_ploidy_cutoff),
+                    'XXX')
+                .or_missing()
+        ),
+        Y_karyotype=(
+            hl.case()
                 .when(
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > yy_y_ploidy_cutoff) &
-                (f_stat > f_stat_male_cutoff),
-                'XYY')
+                    ((ht.chrY_ploidy < xy_y_ploidy_cutoffs[0]) & 
+                    (ht.chrY_ploidy > xx_y_ploidy_cutoff)),
+                    'Y-')
                 .when(
-                (chr_x_ploidy > xx_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xx_x_ploidy_cutoffs[1]) &
-                (chr_y_ploidy > xy_y_ploidy_cutoffs[0]) &
-                (chr_y_ploidy < xy_y_ploidy_cutoffs[1]) &
-                (f_stat < f_stat_female_cutoff),
-                'XXY')
+                    ((ht.chrY_ploidy > xy_y_ploidy_cutoffs[0]) &
+                    (ht.chrY_ploidy < xy_y_ploidy_cutoffs[1])),
+                    'Y')
                 .when(
-                (chr_x_ploidy > xxx_x_ploidy_cutoff) &
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (f_stat < f_stat_female_cutoff),
-                'XXX')
+                    ((ht.chrY_ploidy > xy_y_ploidy_cutoffs[1]) &
+                    (ht.chrY_ploidy < yy_y_ploidy_cutoff)),
+                    'Y+')
                 .when(
-                (chr_y_ploidy < xx_y_ploidy_cutoff) &
-                (chr_x_ploidy > xy_x_ploidy_cutoffs[0]) &
-                (chr_x_ploidy < xy_x_ploidy_cutoffs[1]) &
-                (f_stat > f_stat_male_cutoff),
-                'X0')
-                .default('Ambiguous')
+                     (ht.chrY_ploidy >= yy_y_ploidy_cutoff),
+                    'YY')
+                .or_missing()
+            )
+    )
+
+    sex_expr = sex_expr.annotate(
+        sex_karyotype=(hl.case()
```

Comment:
I wonder if this expression could be simplified if the expressions above used `default('')` instead or `or_missing`. Seems like we could simply do `X_karyotype` +  `Y_karyotype`. Though  you'd need a special case to replace it with `Ambiguous` if it's `len` == 0. Also, this would return `X` rather than `X0`...which I wonder if it is a problem or not -- maybe something to double-check?